debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbChangeEventSourceFactory.java

```diff
@@ -69,11 +70,11 @@ public MongoDbChangeEventSourceFactory(MongoDbConnectorConfig configuration, Err
     }
 
     @Override
-    public Optional<IncrementalSnapshotChangeEventSource<? extends DataCollectionId>> getIncrementalSnapshotChangeEventSource(
-                                                                                                                              MongoDbOffsetContext offsetContext,
-                                                                                                                              SnapshotProgressListener snapshotProgressListener,
-                                                                                                                              DataChangeEventListener dataChangeEventListener) {
-        final MongoDbIncrementalSnapshotChangeEventSource<CollectionId> incrementalSnapshotChangeEventSource = new MongoDbIncrementalSnapshotChangeEventSource<CollectionId>(
+    public Optional<IncrementalSnapshotChangeEventSource<MongoDbPartition, ? extends DataCollectionId>> getIncrementalSnapshotChangeEventSource(
```

Comment:
It would be incompatible with the return type declared by the interface: https://github.com/debezium/debezium/blob/1aa7b7b426864dce077907fe3a006e3236dde023/debezium-core/src/main/java/io/debezium/pipeline/source/spi/ChangeEventSourceFactory.java#L52

I believe if we want the return type to be more specific, it will have to be parametrized more specifically (now even sure how).